src/roadmap/v1/lifecycle/app_streams.py

```diff
@@ -357,54 +325,21 @@ def app_streams_from_modules(
                 impl=AppStreamImplementation.module,
             )
 
-        display_name = get_display_name(matched_module)
-        rolling = get_rolling_value(name, stream, os_major)
-        app_key = AppStreamKey(
-            name=name,
-            display_name=display_name,
-            start_date=matched_module.start_date,
-            end_date=matched_module.end_date,
-            application_stream_name=matched_module.application_stream_name,
-            os_major=os_major,
-            os_minor=os_minor if rolling else None,
-            os_lifecycle=os_lifecycle if rolling else None,
-            rolling=rolling,
-            impl=AppStreamImplementation.module,
-        )
-        app_streams.add(app_key)
+        app_streams.add(matched_module)
 
     return app_streams
 
 
 def app_streams_from_packages(
     package_names_string: str,
     os_major: str,
-    os_minor: str,
-    os_lifecycle: str,
-) -> set[AppStreamKey]:
+) -> set[AppStreamEntity]:
```

Comment:
It looks like defining `__hash__` and only putting in the keys we want to collect on allowed us to get rid of AppStreamKey. That's pretty sweet.